### PR TITLE
docs: update files/dirs for 1.0 release

### PR DESCRIPTION
**Pull request by efiop**
Per https://github.com/iterative/dvc/pull/3725

> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please chose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: content/docs/user-guide/dvc-files-and-directories.md [hunk 1]
```diff
@@ -24,24 +24,28 @@ operation:
   > repository, only [DVC-files](/doc/user-guide/dvc-file-format) that are
   > needed to download or reproduce them.
 
-- `.dvc/state`: This file is used for optimization. It is a SQLite database,
+- `.dvc/tmp`: Directory for miscellaneous temporary files
+
+- `.dvc/tmp/index`: Directory for remote index files that are used for
+  optimizing `dvc push/pull/fetch/status -c` operations.
```
Comment (shcheklein) [inline content/docs/user-guide/dvc-files-and-directories.md:30]: it creates a link automatically to the commands, better split this into more verbose but clean:

`dvc pull -c`, `dvc push -c` etc

Comment (efiop) [inline content/docs/user-guide/dvc-files-and-directories.md:30]:
```suggestion
  `.dvc/tmp/index`: Directory for remote index files that are used for
  optimizing `dvc push`, `dvc pull`, `dvc fetch` and `dvc status -c`
  operations.
```

File: content/docs/user-guide/dvc-files-and-directories.md [hunk 2]
```diff
@@ -24,24 +24,29 @@ operation:
   > repository, only [DVC-files](/doc/user-guide/dvc-file-format) that are
   > needed to download or reproduce them.
 
-- `.dvc/state`: This file is used for optimization. It is a SQLite database,
+- `.dvc/tmp`: Directory for miscellaneous temporary files
+
+- `.dvc/tmp/index`: Directory for remote index files that are used for
+  optimizing `dvc push`, `dvc pull`, `dvc fetch` and `dvc status -c`
+  operations.
```
Comment (efiop) [inline content/docs/user-guide/dvc-files-and-directories.md:31]:
```suggestion
  optimizing `dvc push`, `dvc pull`, `dvc fetch` and `dvc status -c` operations.
```

**Comment by jorgeorpinel** [comment]
Great! Should we create a tag in this repo before any 1.0 changes, so we can release a pdf version of the docs for 0.9? If so, was this the first 1.0-related change? Thanks

File: content/docs/user-guide/dvc-files-and-directories.md [hunk 3]
```diff
@@ -24,24 +24,28 @@ operation:
...
+- `.dvc/tmp/state-journal`: Temporary file for SQLite operations
 
-- `.dvc/state-wal`: Another SQLite temporary file
+- `.dvc/tmp/state-wal`: Another SQLite temporary file
 
-- `.dvc/updater`: This file is used store the latest available version of DVC.
-  It's used to remind the user to upgrade when the installed version is behind.
+- `.dvc/tmp/updater`: This file is used store the latest available version of
+  DVC. It's used to remind the user to upgrade when the installed version is
+  behind.
```
Comment (skshetry) [inline content/docs/user-guide/dvc-files-and-directories.md:44]: Why do we need to document for `.dvc/tmp` directory at all?

Comment (efiop) [reply content/docs/user-guide/dvc-files-and-directories.md:44]: @skshetry Sure, we could make it a black box, but people will still get nosy and will ask questions. So easier to document it once and for all :slightly_smiling_face:

**Pull request merged**